src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts

```diff
@@ -0,0 +1,53 @@
+import { Result } from "@js-soft/ts-utils";
+import { AttributesController } from "@nmshd/consumption";
+import { CoreId } from "@nmshd/transport";
+import { Inject } from "typescript-ioc";
+import { LocalAttributeDTO } from "../../../types";
+import { AttributeIdString, RuntimeErrors, SchemaRepository, SchemaValidator, UseCase } from "../../common";
+import { flattenObject } from "../requests/flattenObject";
+import { AttributeMapper } from "./AttributeMapper";
+import { GetAttributesRequestQuery, GetAttributesUseCase } from "./GetAttributes";
+
+export interface GetSharedVersionsOfIdentityAttributeRequest {
+    id: AttributeIdString;
+}
+
+class Validator extends SchemaValidator<GetSharedVersionsOfIdentityAttributeRequest> {
+    public constructor(@Inject schemaRepository: SchemaRepository) {
+        super(schemaRepository.getSchema("GetSharedVersionsOfIdentityAttributeRequest"));
+    }
+}
+
+export class GetSharedVersionsOfIdentityAttributeUseCase extends UseCase<GetSharedVersionsOfIdentityAttributeRequest, LocalAttributeDTO[]> {
+    public constructor(
+        @Inject private readonly attributeController: AttributesController,
+        @Inject validator: Validator
+    ) {
+        super(validator);
+    }
+
+    protected async executeInternal(request: GetSharedVersionsOfIdentityAttributeRequest): Promise<Result<LocalAttributeDTO[]>> {
+        const attribute = await this.attributeController.getLocalAttribute(CoreId.from(request.id));
+
+        if (!attribute || attribute.isRelationshipAttribute()) {
+            throw RuntimeErrors.general.invalidPropertyValue("The given ID doesn't belong to a valid IdentityAttribute.");
+        }
+
+        const allVersions = await this.attributeController.getVersionsOfAttribute(CoreId.from(request.id));
+
+        // queried ID doesn't belong to a repository attribute
+        if (attribute.shareInfo !== undefined) {
+            return Result.ok(AttributeMapper.toAttributeDTOList(allVersions));
+        }
+
+        const query: GetAttributesRequestQuery = {};
+        query["shareInfo.sourceAttribute"] = allVersions.map((x) => x.id.toString());
+
+        const flattenedQuery = flattenObject(query);
+        const dbQuery = GetAttributesUseCase.queryTranslator.parse(flattenedQuery);
+
+        const sharedAttributes = await this.attributeController.getLocalAttributes(dbQuery);
```

Comment:
We should not use the queryTranslator here, rather following syntax (as db queries):

```suggestion
        const query = {
                "shareInfo.sourceAttribute": { $in: allVersions.map((x) => x.id.toString())}
        };

        const sharedAttributes = await this.attributeController.getLocalAttributes(query);
```